src/data/library.ts

```diff
@@ -151,6 +122,85 @@ class Library {
     });
   };
 
+  // NOTE this is a high-level public API (eposed on ExcalidrawAPI) with
+  // a slight overhead (always restoring library items). For internal use
+  // where merging isn't needed, use `library.setLibrary()` directly.
+  updateLibrary = async ({
+    libraryItems,
+    prompt = false,
```

Comment:
instead of this should we expose a callback, `onBeforeUpdateLibrary` ? so host can handle it and customize it accordingly

Reply:
would complicate things a lot. Maybe at some point we can do it, but so far it seems unnecessary. Up to this point we've been handling it ourselves, so let's keep it that way, and potentially decide on changing this in another PR.

Reply:
`prompt` attribute is newly introduced in this version if I remember correctly? API wise I would try to avoid it and keep the default behavior instead(always show prompt), and in later PR we can introduce customization if needed. Adding a `prompt` attribute doesn't look like a good API so lets not avoid it if possible.

Reply:
I see. This is to keep the current behavior. 

Library UI is still handled by us (and even later we'll just factor it out to a reusable component). Forcing host apps to handle this part, which is kinda integral to it, is not good (especially once we move to an on-brand confirmation dialog). Also they'd lose i18n.

We can rename to `askForConfirmation`.